src/main/java/de/tum/in/www1/artemis/service/competency/StandardizedCompetencyService.java

```diff
@@ -0,0 +1,72 @@
+
+import javax.ws.rs.BadRequestException;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.competency.KnowledgeArea;
+import de.tum.in.www1.artemis.domain.competency.Source;
+import de.tum.in.www1.artemis.domain.competency.StandardizedCompetency;
+import de.tum.in.www1.artemis.repository.SourceRepository;
+import de.tum.in.www1.artemis.repository.competency.KnowledgeAreaRepository;
+import de.tum.in.www1.artemis.repository.competency.StandardizedCompetencyRepository;
+
+/**
+ * Service for managing {@link StandardizedCompetency} entities.
+ */
+@Profile(PROFILE_CORE)
+@Service
+public class StandardizedCompetencyService {
+
+    private static final String FIRST_VERSION = "1.0.0";
+
+    private final StandardizedCompetencyRepository standardizedCompetencyRepository;
+
+    private final KnowledgeAreaRepository knowledgeAreaRepository;
+
+    private final SourceRepository sourceRepository;
+
+    public StandardizedCompetencyService(StandardizedCompetencyRepository standardizedCompetencyRepository, KnowledgeAreaRepository knowledgeAreaRepository,
+            SourceRepository sourceRepository) {
+        this.standardizedCompetencyRepository = standardizedCompetencyRepository;
+        this.knowledgeAreaRepository = knowledgeAreaRepository;
+        this.sourceRepository = sourceRepository;
+    }
+
+    public StandardizedCompetency createStandardizedCompetency(StandardizedCompetency competency) {
+        standardizedCompetencyIsValidOrElseThrow(competency);
+
+        // fetch the knowledge area and source from the database if they exists
+        KnowledgeArea knowledgeArea = competency.getKnowledgeArea();
+        if (knowledgeArea != null) {
+            knowledgeArea = knowledgeAreaRepository.findByIdElseThrow(knowledgeArea.getId());
+        }
+        Source source = competency.getSource();
+        if (source != null) {
+            source = sourceRepository.findByIdElseThrow(source.getId());
+        }
+
+        var competencyToCreate = new StandardizedCompetency(competency.getTitle().trim(), competency.getDescription(), competency.getTaxonomy(), FIRST_VERSION);
+        competencyToCreate.setKnowledgeArea(knowledgeArea);
+        competencyToCreate.setSource(source);
+
+        competencyToCreate = standardizedCompetencyRepository.save(competencyToCreate);
+        // set the reference to the first version of this standardized competency (itself)
+        competencyToCreate.setFirstVersion(competencyToCreate);
+        return standardizedCompetencyRepository.save(competencyToCreate);
+    }
+
+    /**
+     * Verifies that a standardized competency that should be created is valid or throws a BadRequestException
+     *
+     * @param competency the standardized competency to verify
+     */
+    public void standardizedCompetencyIsValidOrElseThrow(StandardizedCompetency competency) throws BadRequestException {
```

Comment:
Should this be public? Any use of this method outside of this service would indicate that the logic should probably be part of it.